src/components/TransactionBlock/HoneySwapLPTransactionBlock.tsx

```diff
@@ -0,0 +1,492 @@
+  const [selectedToken1Asset, setSelectedToken1Asset] = useState<IAssetWithBalance | null>(values?.toToken1 ?? null);
+  const [selectedToken2Asset, setSelectedToken2Asset] = useState<IAssetWithBalance | null>(values?.toToken2 ?? null);
+
+  const defaultCustomReceiverAddress =
+    values?.receiverAddress &&
+    !addressesEqual(providerAddress, values?.receiverAddress) &&
+    !addressesEqual(accountAddress, values?.receiverAddress)
+      ? values.receiverAddress
+      : null;
+  const [customReceiverAddress, setCustomReceiverAddress] = useState<string | null>(defaultCustomReceiverAddress);
+  const [useCustomAddress, setUseCustomAddress] = useState<boolean>(!!defaultCustomReceiverAddress);
+
+  const defaultSelectedReceiveAccountType =
+    (!values?.receiverAddress && values?.accountType === AccountTypes.Key) ||
+    (values?.receiverAddress &&
+      values?.accountType === AccountTypes.Contract &&
+      addressesEqual(providerAddress, values?.receiverAddress))
+      ? AccountTypes.Key
+      : AccountTypes.Contract;
+  const [selectedReceiveAccountType, setSelectedReceiveAccountType] = useState<string>(
+    defaultSelectedReceiveAccountType
+  );
+
+  const {
+    setTransactionBlockValues,
+    resetTransactionBlockFieldValidationError,
+    setTransactionBlockFieldValidationError,
+  } = useTransactionBuilder();
+
+  const theme: Theme = useTheme();
+
+  useEffect(() => {
+    resetTransactionBlockFieldValidationError(transactionBlockId, 'amount');
+  }, [selectedFromNetwork]);
+
+  const onAmountChange = useCallback(
+    (newAmount: string) => {
+      resetTransactionBlockFieldValidationError(transactionBlockId, 'amount');
+      const decimals = selectedFromAsset?.decimals ?? 18;
+      const updatedAmount = formatAssetAmountInput(newAmount, decimals);
+      setAmount(updatedAmount);
+    },
+    [selectedFromAsset]
+  );
+
+  const receiverAddress = useMemo(() => {
+    if (useCustomAddress) return customReceiverAddress;
+    return selectedReceiveAccountType === AccountTypes.Key ? providerAddress : accountAddress;
+  }, [
+    useCustomAddress,
+    customReceiverAddress,
+    providerAddress,
+    selectedReceiveAccountType,
+    selectedAccountType,
+    accountAddress,
+  ]);
+
+  useEffect(() => {
+    if (selectedFromAsset?.assetPriceUsd && +amount * selectedFromAsset.assetPriceUsd < 0.4) {
+      setTransactionBlockFieldValidationError(transactionBlockId, 'amount', 'Minimum amount 0.4 USD');
+      return;
+    }
+    resetTransactionBlockFieldValidationError(transactionBlockId, 'amount');
+    if (receiverAddress && !isValidEthereumAddress(receiverAddress)) {
+      setTransactionBlockFieldValidationError(transactionBlockId, 'receiverAddress', 'Invalid receiver address');
+      return;
+    }
+
+    if (routeToUSDC.length == 0) {
+      setTransactionBlockFieldValidationError(transactionBlockId, 'route', 'Please try with different inputs/amount');
+      return;
+    }
```

Comment:
It''s step by step in case the first one is invalid the user needs to fix the first, in case the second one the user needs to do the same with the second one etc.